MusicApp.SongService.Infrastructure/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.EntityFrameworkCore;
+using MusicApp.SongService.Application.Repositories;
+using MusicApp.SongService.Infrastructure.Data;
+
+namespace MusicApp.SongService.Infrastructure.Repositories;
+
+public abstract class BaseRepository<T> : IBaseRepository<T> where T : class
+{
+    private readonly AppDbContext _db;
+    private readonly DbSet<T> _dbSet;
+
+    public BaseRepository(AppDbContext db)
+    {
+        _db = db;
+        _dbSet = _db.Set<T>();
+    }
+
+    public virtual async Task<IEnumerable<T>> GetAsync(CancellationToken cancellationToken)
+    {
+        return await _dbSet.AsNoTracking().ToListAsync(cancellationToken);
+    }
+
+    public virtual async Task<T> GetByIdAsync(Guid id, CancellationToken cancellationToken)
+    {
+        return await _dbSet.FindAsync(new object[] { id }, cancellationToken);
```

Comment:
Why find? And this object array... It's strange

Reply:
Because in the BaseRepository I have a generic type as class and I cannot use FirstOrDefaultAsync because the class does not have an id property defined. 
I can create a parent class Entity that will have an id property defined and write this method via FindFirstOrDefault.